src/gpio.rs

```diff
@@ -50,1205 +73,1450 @@ pub trait GpioExt {
-                Ok(!self.is_low()?)
-            }
+#[cfg(feature = "unproven")]
+impl<GPIO, INDEX, MODE> InputPin for Pin<GPIO, INDEX, MODE>
+where
+    GPIO: Gpio,
+    INDEX: Index,
+    MODE: Readable,
+{
+    type Error = Infallible;
+
+    fn is_high(&self) -> Result<bool, Self::Error> {
+        Ok(!self.is_low()?)
+    }
+
+    fn is_low(&self) -> Result<bool, Self::Error> {
+        // NOTE(unsafe) atomic read with no side effects
+        Ok(unsafe { (*self.gpio.ptr()).is_low(self.index.index()) })
+    }
+}
 
-             fn is_low(&self) -> Result<bool, Self::Error> {
-                // NOTE(unsafe) atomic read with no side effects
-                Ok(unsafe {
-                    match &self.gpio {
-                        $(
-                            Gpio::$GPIOX => (*$GPIOX::ptr()).idr.read().bits() & (1 << self.i) == 0,
-                        )+
-                    }
-                })
-            }
-        }
+#[cfg(feature = "unproven")]
+impl<GPIO, INDEX, OTYPE> StatefulOutputPin for Pin<GPIO, INDEX, Output<OTYPE>>
+where
+    GPIO: Gpio,
+    INDEX: Index,
+{
+    fn is_set_high(&self) -> Result<bool, Self::Error> {
+        Ok(!self.is_set_low()?)
+    }
+
+    fn is_set_low(&self) -> Result<bool, Self::Error> {
+        // NOTE(unsafe) atomic read with no side effects
+        Ok(unsafe { (*self.gpio.ptr()).is_set_low(self.index.index()) })
+    }
+}
 
-        #[cfg(feature = "unproven")]
-        impl InputPin for PXx<Output<OpenDrain>> {
-            type Error = Infallible;
+#[cfg(feature = "unproven")]
+impl<GPIO, INDEX, OTYPE> toggleable::Default for Pin<GPIO, INDEX, Output<OTYPE>>
+where
+    GPIO: Gpio,
+    INDEX: Index,
+{
+}
 
-            fn is_high(&self) -> Result<bool, Self::Error> {
-                Ok(!self.is_low()?)
+impl<GPIO, INDEX, MODE> Pin<GPIO, INDEX, MODE>
+where
+    GPIO: Gpio,
+    INDEX: Index,
+    MODE: Active,
+{
+    /// Make corresponding EXTI line sensitive to this pin
+    pub fn make_interrupt_source(&mut self, syscfg: &mut SysCfg) {
+        let i = self.index.index() % 4;
+        let extigpionr = self.gpio.port_index() as u32;
+        match self.index.index() {
+            0..=3 => unsafe { modify_at!(syscfg.exticr1, 4, i, extigpionr) },
+            4..=7 => unsafe { modify_at!(syscfg.exticr2, 4, i, extigpionr) },
+            8..=11 => unsafe { modify_at!(syscfg.exticr3, 4, i, extigpionr) },
+            12..=15 => unsafe { modify_at!(syscfg.exticr4, 4, i, extigpionr) },
+            _ => unreachable!(),
+        }
+    }
+
+    /// Generate interrupt on rising edge, falling edge, or both
+    pub fn trigger_on_edge(&mut self, exti: &mut EXTI, edge: Edge) {
+        cfg_if! {
+            // if #[cfg(any(feature = "stm32f373", feature = "stm32f378"))] { // TODO: update after https://github.com/stm32-rs/stm32-rs/pull/496 released
+            if #[cfg(not(any(feature = "stm32f302", feature = "stm32f303", feature = "stm32f334")))] {
```

Comment:
I am not sure this conditional compilation is correct. Looking at the [stm32-rs/STM32F301](https://stm32-rs.github.io/stm32-rs/STM32F301.html#EXTI:RTSR1). It does not fall in this feature category but does have a numbered RTSR.

Reply:
The [stm32-rs/STM32F301](https://stm32-rs.github.io/stm32-rs/STM32F301.html#EXTI:RTSR1) is based on latest nightly version of stm32-rs, while we are using v0.12 for now.